### PR TITLE
Add coverage to make file

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following introduces code coverage to the tests, this should allow
us to see what's missing and what we can remove in terms of tests
without loss to the project.
## QA steps

```sh
COVERAGE_CHECK=1 make run-tests
```

### CODE REVIEW

File: .github/workflows/client-tests.yml [hunk 1]
```diff
@@ -98,6 +98,10 @@ jobs:
       if: (matrix.os == 'ubuntu-latest')
       run: |
         # Jenkins can perform the full jujud testing.
-        go test -v ./cmd/juju/... -check.v
+        go test -v ./cmd/juju/... -check.v -coverprofile=coverage.txt -covermode=atomic
```
Comment (achilleasa) [inline .github/workflows/client-tests.yml:101]: Have we verified that running this both as a GH action and on jenkins for the same SHA does not clobber the coverage results?

Comment (SimonRichardson) [reply .github/workflows/client-tests.yml:101]: Not yet, but consider this is only an experiment, we can fix that if we move it to jenkins.

**Review by achilleasa — APPROVED** [review]
Go for it

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by petevg** [comment]
Merging, because the failed merge seems to be a bug in the plugin we're using rather than a defect in the code.

**Pull request merged**